src/hotspot/cpu/aarch64/aarch64_sve.ad

```diff
@@ -340,6 +351,62 @@ instruct storeV_partial(vReg src, vmemA mem, pRegGov pTmp, iRegINoSp tmp1,
   %}
   ins_pipe(pipe_slow);
 %}
+// vector reinterpret
+
+instruct reinterpret(vReg dst) %{
+  predicate(UseSVE > 0 && n->as_Vector()->length_in_bytes() >= 16 &&
+            n->as_Vector()->length_in_bytes() ==
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes());  // src == dst
+  match(Set dst (VectorReinterpret dst));
+  ins_cost(0);
+  format %{ " # reinterpret $dst" %}
+  ins_encode %{
+    // empty
+  %}
+  ins_pipe(pipe_class_empty);
+%}
+
+instruct reinterpretExpand(vReg dst, vReg src, iRegINoSp tmp, pRegGov pTmp, rFlagsReg cr) %{
+  predicate(UseSVE > 0 && n->in(1)->bottom_type()->is_vect()->length_in_bytes() >= 16 &&
+            n->as_Vector()->length_in_bytes() >
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes());  // src < dst
+  match(Set dst (VectorReinterpret src));
+  effect(TEMP pTmp, TEMP tmp, KILL cr);
+  ins_cost(2 * SVE_COST);
+  format %{ " # reinterpretExpand $dst,$src\t" %}
+  ins_encode %{
+    // If register is the same, then move is not needed.
+    if (as_FloatRegister($dst$$reg) != as_FloatRegister($src$$reg)) {
+      __ mov(as_Register($tmp$$reg), vector_length_in_bytes(this, $src));
+      __ sve_whilelo(as_PRegister($pTmp$$reg), __ B,
+                     zr, as_Register($tmp$$reg));
+      __ sve_cpy(as_FloatRegister($dst$$reg), __ B,
+                 as_PRegister($pTmp$$reg), as_FloatRegister($src$$reg));
```

Comment:
It seems using `cpy` is not right here, which usually accept a `imm, scalar or SIMD&FP scalar register`. This instruction is expected to copy the scalar value to each active element. So use `"mov"` instead ? BTW, the normal `MOV <Zd>.<T>, <Pg>/M, <Zn>.<T>`  instruction uses the `merging` feature of predicate while not `zeroing`. To make sure the non active elements are zero, do you think it's better to use `MOVS <Pd>.B, <Pg>/Z, <Pn>.B` here?

Reply:
I'm sorry that I didn't notice this is an instruction for predicate `MOVS <Pd>.B, <Pg>/Z, <Pn>.B`. So please ignore this suggestion. Thanks！

Reply:
OK. We change our algorithm and use other instructions instead of these.